content/automate/ManageIQ/System/Event/StateMachines/Refresh.class/__methods__/check_refreshed.rb

```diff
@@ -0,0 +1,59 @@
+#
+# Description: This method checks the refresh completion.
+#
+
+module ManageIQ
+  module Automate
+    module System
+      module Event
+        module StateMachines
+          module Refresh
+            class CheckRefreshed
+              STATE_FINISHED = 'Finished'.freeze
+              STATUS_OK = 'Ok'.freeze
+
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                check_status(refresh_task)
+              end
+
+              private
+
+              def check_status(task)
+                case task.state
+                when STATE_FINISHED
+                  @handle.root['ae_result'] = task.status == STATUS_OK ? 'ok' : 'error'
```

Comment:
@lfu 
If there is an error we should show it in this log. MiqTask#message would have the error message.